src/apps/components/AppFrame/appActionsHandler.ts

```diff
@@ -0,0 +1,187 @@
+import { useExternalApp } from "@dashboard/apps/components/ExternalAppContext/ExternalAppContext";
+import { getAppMountUri } from "@dashboard/config";
+import useNavigator from "@dashboard/hooks/useNavigator";
+import useNotifier from "@dashboard/hooks/useNotifier";
+import { AppUrls } from "@dashboard/new-apps/urls";
+import {
+  DispatchResponseEvent,
+  NotificationAction,
+  NotifyReady,
+  RedirectAction,
+  UpdateRouting,
+} from "@saleor/app-sdk/app-bridge";
+import { useIntl } from "react-intl";
+import urlJoin from "url-join";
+
+const createResponseStatus = (
+  actionId: string,
+  ok: boolean,
+): DispatchResponseEvent => ({
+  type: "response",
+  payload: {
+    actionId,
+    ok,
+  },
+});
+
+const isExternalHost = (host: string) =>
+  new URL(host).hostname !== window.location.hostname;
+
+const isLocalPath = (path: string) => path.startsWith("/");
+
+const useHandleNotificationAction = () => {
+  const notify = useNotifier();
+
+  return {
+    handle: (action: NotificationAction) => {
+      const { actionId, ...notification } = action.payload;
+
+      console.debug(`Handling Notification action with ID: ${actionId}`);
+
+      notify({
+        ...notification,
+      });
+
+      return createResponseStatus(actionId, true);
+    },
+  };
+};
+
+const useHandleRedirectAction = (appId: string) => {
+  const navigate = useNavigator();
+  const { closeApp } = useExternalApp();
+  const intl = useIntl();
+
+  const handleAppDeepChange = (action: RedirectAction) => {
+    console.debug("Handling deep app URL change");
+    const exactLocation = urlJoin(getAppMountUri(), action.payload.to);
+    console.debug(`Exact location to redirect: ${exactLocation}`);
```

Comment:
**question**: Should it affect normal dashboard usage? 

Is using debug logging to the console in production a good idea if the end user could see those logs? If so maybe it will be worth adding such logging to sentry?

Reply:
I dont think it affects
1. `debug` logs are hidden by default
2. All of these information an be traced anyway in debugger and this is open source
3. app-sdk is also logging (there is more sophisticated debug, enabled in localStorage flag)
4. Sentry is not logging platform. We should have logs in datadog or something. But Sentry only catches errors (https://sentry.io/vs/logging/)

I think it should not be removed in production because we can... debug. But if this is something against dashboard approach, I can remove them until you figure out some logging system

WDYT ? 

cc @andrzejewsky

Reply:
We need logging anyway, the project is quite big and very often it's hard to figure out what's happening. We can keep console.debug for now if that helps, when later on we introduce something that cover it (perhaps something that is already adopted in our ecosystem)